Documentation/BasicOperators.md

```diff
@@ -49,68 +49,72 @@ types will be referred to by name.
 
 ### Observation
 
-`Signal`s can be observed with the `observe` function. It takes an `Observer` as argument to which any future events are sent. 
+`Signal`s can be observed with the `observe` function.
 
 ```Swift
-signal.observe(Signal.Observer { event in
+signal.observe { event in
     switch event {
-    case let .Next(next):
-        print("Next: \(next)")
-    case let .Failed(error):
+    case let .value(value):
+        print("Value: \(value)")
+    case let .failed(error):
         print("Failed: \(error)")
-    case .Completed:
+    case .completed:
         print("Completed")
-    case .Interrupted:
+    case .interrupted:
         print("Interrupted")
     }
-})
+}
 ```
 
-Alternatively, callbacks for the `Next`, `Failed`, `Completed` and `Interrupted` events can be provided which will be called when a corresponding event occurs.
+Alternatively, callbacks for the `value`, `failed`, `completed` and `interrupted` events can be provided which will be called when a corresponding event occurs.
 
 ```Swift
-signal.observeNext { next in 
-  print("Next: \(next)") 
+signal.observeValues { value in
+    print("Value: \(value)")
 }
+
 signal.observeFailed { error in
-  print("Failed: \(error)")
+    print("Failed: \(error)")
 }
-signal.observeCompleted { 
-  print("Completed") 
+
+signal.observeCompleted {
+    print("Completed")
 }
-signal.observeInterrupted { 
-  print("Interrupted")
+
+signal.observeInterrupted {
+    print("Interrupted")
 }
 ```
 
-Note that it is not necessary to observe all four types of event - all of them are optional, you only need to provide callbacks for the events you care about.
-
 ### Injecting effects
 
-Side effects can be injected on a `SignalProducer` with the `on` operator without actually subscribing to it. 
+Side effects can be injected on an event stream with the `on` operator without actually subscribing to it. 
 
 ```Swift
 let producer = signalProducer
-    .on(started: {
+    .on(starting: { 
+        print("Starting")
+    }, started: { 
         print("Started")
     }, event: { event in
         print("Event: \(event)")
+    }, value: { value in
+        print("Value: \(value)")
     }, failed: { error in
         print("Failed: \(error)")
-    }, completed: {
+    }, completed: { 
         print("Completed")
-    }, interrupted: {
+    }, interrupted: { 
         print("Interrupted")
-    }, terminated: {
+    }, terminated: { 
         print("Terminated")
-    }, disposed: {
+    }, disposed: { 
         print("Disposed")
-    }, next: { value in
```

Comment:
You are talking about this line: 
> Note that nothing will be printed until `producer` is started (possibly somewhere else).

right?

Sorry, I'm not sure what you mean exactly. I've tried this example in a Playground
```
let signalProducer = SignalProducer<Int, NoError>(value: 1)

let producer = signalProducer
    .on(starting: {
        print("Starting")
    }, started: {
        print("Started")
    }, event: { event in
        print("Event: \(event)")
    }, value: { value in
        print("Value: \(value)")
    }, failed: { error in
        print("Failed: \(error)")
    }, completed: {
        print("Completed")
    }, interrupted: {
        print("Interrupted")
    }, terminated: {
        print("Terminated")
    }, disposed: {
        print("Disposed")
    })

print("---")
producer.start()
print("---")
```

And it behaved as expected and as the line says ("nothing is printed until the producer is started"):

```
---
Starting
Event: VALUE 1
Value: 1
Event: COMPLETED
Completed
Terminated
Disposed
Started
---
```